### PR TITLE
edit build system ref topic

### PR DESCRIPTION
**Pull request by guillermooo**

### CODE REVIEW

File: source/reference/build_systems.rst [hunk 1]
```diff
@@ -10,37 +10,44 @@ and see the output they generate.
...
-the data stored in the *.sublime-build* file.
-At this point, it can do whatever it needs to do,
+When you run a build action in Sublime Text
+(usually, by pressing :kbd:`Ctrl+B`),
+a Sublime Text command receives
+this configuration data.
+The command then does whatever it needs
 to *build* the files.
+Often, it forwards configuration data
+to an external program.
```
Comment (FichteFoll) [inline source/reference/build_systems.rst]: I don't like that phrasing. Configuration data is not exactly "forwarded", aside from the command parameters. Everything else is for ST internally.

File: source/reference/build_systems.rst [hunk 2]
```diff
@@ -10,37 +10,44 @@ and see the output they generate.
...
+
 By default, build systems will use
 the ``exec`` command implemented by :file:`Packages/Default/exec.py`.
-As we'll explain below,
-you can override this command.
+As we'll explain below, however,
+you this command can be overriden.
```
Comment (FichteFoll) [inline source/reference/build_systems.rst]: needs fixing

File: source/reference/build_systems.rst [hunk 3]
```diff
@@ -10,37 +10,44 @@ and see the output they generate.
...
+receives paths to files or directories,
+along with switches and options
+that configure its behavior.
 
 Note that build systems can but don't need to
 call external programs;
-a valid build system could be implemented entirely
-in Python in a Sublime Text command.
+a build system could be implemented entirely
+as a Sublime Text command.
```
Comment (FichteFoll) [inline source/reference/build_systems.rst]: From what I remember, it must be a `WindowCommand`, but better ensure it's true before. Where should we mention that?

File: source/reference/build_systems.rst [hunk 4]
```diff
@@ -84,28 +91,32 @@ in the *.sublime-build* file.
     for the active view.
 
 ``windows``, ``osx`` and ``linux``
-    Optional. Allow specification of OS-specific options
-    which will override the default settings.
-    These accept a dict of `Arbitrary options`_ each.
+    Optional. Used to selectively apply options by OS.
+    OS-specific values override defaults.
+    Each of the listed items accepts a dict of `Arbitrary options`_.
```
Comment (FichteFoll) [inline source/reference/build_systems.rst]: "dict" vs "dictionary"? (dictionary used below)

File: source/reference/build_systems.rst [hunk 5]
```diff
@@ -84,28 +91,32 @@ in the *.sublime-build* file.
...
+    Using variants it's possible
+    to specify multiple build systems
+    for different tasks
+    in the same *.sublime-build* file.
+
     See Variants_.
 
 ``name``
-    **Only valid inside a variant** (see ``variants``).
+    **Only valid inside a variant** (see Variants_).
```
Comment (FichteFoll) [inline source/reference/build_systems.rst]: Add a line break after this.

File: source/reference/build_systems.rst [hunk 6]
```diff
@@ -84,28 +91,32 @@ in the *.sublime-build* file.
...
     Identifies variant build systems.
-    If ``name`` is *Run*,
+    If the ``name`` is *Run*,
     the variant will show up
     under the **Tools | Build System** menu
-    and be bound to :kbd:`Ctrl+Shift+B`.
+    and be bound to :kbd:`Ctrl+B`.
```
Comment (FichteFoll) [inline source/reference/build_systems.rst]:
```
{ "keys": ["ctrl+b"], "command": "build" },
{ "keys": ["ctrl+shift+b"], "command": "build", "args": {"variant": "Run"} },
```

**Comment by FichteFoll** [comment]
Along with the other things, you didn't follow (or didn't adjust to) our guidelines for menu items, file extensions etc.

And finally, as I noted in the corresponding [trello card](https://trello.com/c/UxRz3eyW/15-build-systems), I'm not too fond of the name "Arbitrary options" anymore, because it's rather misleading. They should just be renamed to something like "`exec` Command Parameters".

**Comment by guillermooo** [comment]
> Along with the other things, you didn't follow (or didn't adjust to) our guidelines for menu items, file extensions etc.

On the first pass, I'm focusing on content, but yeah.

> And finally, as I noted in the corresponding trello card, I'm not too fond of the name "Arbitrary options" anymore, because it's rather misleading. They should just be renamed to something like "exec Command Parameters".

I'll take a look at that.

File: source/reference/build_systems.rst (hunk 1 above)
Comment (guillermooo) [inline source/reference/build_systems.rst]: Done.

File: source/reference/build_systems.rst (hunk 2 above)
Comment (guillermooo) [inline source/reference/build_systems.rst]: Done.

File: source/reference/build_systems.rst (hunk 3 above)
Comment (guillermooo) [inline source/reference/build_systems.rst]: Done.

File: source/reference/build_systems.rst (hunk 4 above)
Comment (guillermooo) [inline source/reference/build_systems.rst]: Done.

File: source/reference/build_systems.rst (hunk 5 above)
Comment (guillermooo) [inline source/reference/build_systems.rst]: Done.

**Comment by guillermooo** [comment]
- [x] review trello card

**Comment by guillermooo** [comment]
@FichteFoll PTAL. Tried to address the points you mentioned.

**Comment by guillermooo** [comment]
@FichteFoll I found the topic too long, so I've split it up into subtopics. I think it's more manageable like this, but let me know what you think.

**Comment by FichteFoll** [comment]
I actually meant a single newline after "Optional." and "Required." because those are terminated by a full stop.

Instead of dedicating a whole paragraph for optional or required, how about adding that to the definition term instead?

```
``syntax`` *(optional)*
   If provided,
   it will be used to colorize
   the build system's output.
```

**Pull request merged**